clippy_lints/src/lifetimes.rs

```diff
@@ -586,11 +591,46 @@ where
         self.generic_args_depth -= 1;
     }
 
+    fn visit_fn_decl(&mut self, fd: &'tcx FnDecl<'tcx>) -> Self::Result {
+        self.lifetime_elision_impossible = !is_candidate_for_elision(fd);
```

Comment:
I like this way of factoring it out into a function. It makes it more readable, good job :D